src/xenia/base/byte_stream.h

```diff
@@ -0,0 +1,91 @@
+/**
+ ******************************************************************************
+ * Xenia : Xbox 360 Emulator Research Project                                 *
+ ******************************************************************************
+ * Copyright 2015 Ben Vanik. All rights reserved.                             *
+ * Released under the BSD license - see LICENSE in the root for more details. *
+ ******************************************************************************
+ */
+
+#ifndef XENIA_BASE_BYTE_STREAM_H_
+#define XENIA_BASE_BYTE_STREAM_H_
+
+#include <cstdint>
+#include <string>
+
+namespace xe {
+
+class ByteStream {
```

Comment:
nice type - we may be able to replace some of the other types with this (StringReader, Arena, etc)